pkg/orchestrator/config/config.go

```diff
@@ -157,3 +159,23 @@ func extractOrchestratorDDUrl() (*url.URL, error) {
 	}
 	return URL, nil
 }
+
+// NewOrchestratorForwarder returns an orchestratorForwarder
+// if the feature is activated on the cluster-agent/cluster-check runner, nil otherwise
+func NewOrchestratorForwarder(confPath string) *forwarder.DefaultForwarder {
+	if !config.Datadog.GetBool("orchestrator_explorer.enabled") {
+		return nil
+	}
+	if flavor.GetFlavor() == flavor.DefaultAgent && !config.IsCLCRunner() {
+		return nil
+	}
+	orchestratorCfg := NewDefaultOrchestratorConfig()
+	if err := orchestratorCfg.LoadYamlConfig(confPath); err != nil {
```

Comment:
I'm unsure of the side effects of resolving secrets twice on `config.Datadog` in the same process, which is what happens in `LoadYamlConfig` (`ResolveSecrets`). There might be side effects.

Could `LoadYamlConfig` take an initialized `config.Datadog` struct instead?

Reply:
that's a fair point I'm sharing this function with the process-agent but it looks like `config.LoadWithoutSecret()` is used in the later 👀